### PR TITLE
Use Vector to replace new and delete in gint，spilt gint_tools and change Makefile,remove ORB in gint

### PR DESCRIPTION
**Pull request by A-006**
In `MeshK`, `Meshcell`, `bigcell`, `meshball`, and `grid_technique`, use `vector` instead of `new` and `delete`. Some parameters should be protected using `private`. gint_tools was spilted into 6 files,and must modify the Makefile.Objects here, otherwise there will be issues. Remove the dependency on the ORB parameter in gint; the ORB structure is not needed during the gint calculation process.

### CODE REVIEW

**Review by goodchong — APPROVED** [review]
A good step. But future changes like this will need to be carefully examined for performance impact. I tried to solve the problem with some elegant class wrappers, but this had severe performance impacts on some variables.

File: source/module_hamilt_lcao/hamilt_lcaodft/LCAO_nnr.cpp [hunk 1]
```diff
@@ -18,19 +19,14 @@ void Grid_Technique::cal_nnrg(Parallel_Orbitals* pv)
 
 	this->nnrg = 0;
 
-	delete[] nlocdimg;
-	delete[] nlocstartg;
-	delete[] nad;
+	nlocdimg.clear();
+	nlocstartg.clear();
+	nad.clear();
```
Comment (dzzz2001) [inline source/module_hamilt_lcao/hamilt_lcaodft/LCAO_nnr.cpp:24]: too many adundant clear().

File: source/module_hamilt_lcao/module_gint/grid_bigcell.cpp [hunk 2]
```diff
@@ -28,13 +25,9 @@ Grid_BigCell::~Grid_BigCell()
...
 	{
-		for(int i=0; i<this->nat; i++)
-		{
-			delete[] tau_in_bigcell[i];
-		}
-		delete[] tau_in_bigcell;
+		tau_in_bigcell.clear();
 	}
-	delete[] index_atom;
+	index_atom.clear();
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/grid_bigcell.cpp]: In C++, the `clear()` function of `std::vector` is used to remove all elements from the container, which reduces the size of the `vector` to 0. However, calling `clear()` does not change the capacity of the `vector`, which is the amount of allocated space in memory. If you want to release memory, you can use the `shrink_to_fit` function or the swap trick.

Comment (caic99) [reply source/module_hamilt_lcao/module_gint/grid_bigcell.cpp:30]: > If you want to release memory

`resize(0)` would be just fine

**Review by mohanchen — APPROVED** [review]

File: source/module_hamilt_lcao/hamilt_lcaodft/LCAO_nnr.cpp (hunk 1 above)
Comment (A-006) [reply source/module_hamilt_lcao/hamilt_lcaodft/LCAO_nnr.cpp:24]: has been finished

File: source/module_hamilt_lcao/module_gint/grid_bigcell.cpp (hunk 2 above)
Comment (A-006) [reply source/module_hamilt_lcao/module_gint/grid_bigcell.cpp]: It has already been resolved.

File: source/module_esolver/esolver_ks_lcao_elec.cpp [hunk 3]
```diff
@@ -52,6 +52,47 @@ void ESolver_KS_LCAO<TK, TR>::set_matrix_grid(Record_adj& ra)
     // ModuleBase::GlobalFunc::DONE(GlobalV::ofs_running,"SEARCH ADJACENT ATOMS");
 
     // (3) Periodic condition search for each grid.
+    double dr_uniform;
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao_elec.cpp]: put these new codes in esolver_ks_lcao_elec as an independent source file (.cpp)

File: source/module_hamilt_lcao/module_gint/grid_technique.h [hunk 4]
```diff
@@ -20,98 +20,92 @@ class Grid_Technique : public Grid_MeshBall
...
-                      const int& startz_current,
-                      const UnitCell& ucell,
-                      const LCAO_Orbitals& orb,
-                      const int num_stream);
+    void set_pbc_grid(const int& ncx_in, const int& ncy_in, const int& ncz_in, const int& bx_in, const int& by_in,
+                      const int& bz_in, const int& nbx_in, const int& nby_in, const int& nbz_in, const int& nbxx_in,
+                      const int& nbzp_start_in, const int& nbzp_in, const int& ny, const int& nplane,
+                      const int& startz_current, const UnitCell& ucell, const double dr_uniform, 
+                      std::vector<double> rcuts,std::vector<std::vector<double>> psi_u,
+                      std::vector<std::vector<double>> dpsi_u,std::vector<std::vector<double>> d2psi_u
```
Comment (dzzz2001) [inline source/module_hamilt_lcao/module_gint/grid_technique.h]: maybe `rcuts`, `psi_u`, `dpsi_u` and `d2psi_u` should be passed by reference here.

Comment (A-006) [reply source/module_hamilt_lcao/module_gint/grid_technique.h]: It has already been resolved.

File: source/Makefile.Objects [hunk 5]
```diff
@@ -222,6 +222,7 @@ OBJS_ESOLVER_LCAO=esolver_ks_lcao.o\
       esolver_ks_lcao_tddft.o\
       esolver_ks_lcao_tmpfunc.o\
       io_npz.o\
+      orb_paramter_init.o\
```
Comment (mohanchen) [inline source/Makefile.Objects]: put this file into gint module.

File: source/module_esolver/esolver_ks_lcao.h [hunk 6]
```diff
@@ -109,6 +109,13 @@ namespace ModuleESolver
         // set matrix and grid integral
         void set_matrix_grid(Record_adj& ra);
 
+        void orb_paramter_init(double& dr_uniform,
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao.h]: make this an independent function, not a member function of esolver.

File: source/module_esolver/esolver_ks_lcao_elec.cpp [hunk 7]
```diff
@@ -52,6 +52,15 @@ void ESolver_KS_LCAO<TK, TR>::set_matrix_grid(Record_adj& ra)
...
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao_elec.cpp]: initialize the value

File: source/module_esolver/orb_paramter_init.cpp [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+                                std::vector<double>& rcuts,
+                                UnitCell& ucell,
+                                std::vector<std::vector<double>>& psi_u,
+                                std::vector<std::vector<double>>& dpsi_u,
+                                std::vector<std::vector<double>>& d2psi_u)
+{
+    // set the grid parameters
+    dr_uniform=GlobalC::ORB.dr_uniform;
+    
+    int nwmax=ucell.nwmax;
```
Comment (mohanchen) [inline source/module_esolver/orb_paramter_init.cpp]: add const if possible

File: source/Makefile.Objects (hunk 5 above)
Comment (A-006) [reply source/Makefile.Objects]: This function must use the GloablC parameter, so it is more appropriate to place it in the ORB folder.

File: source/module_esolver/esolver_ks_lcao_elec.cpp (hunk 7 above)
Comment (A-006) [reply source/module_esolver/esolver_ks_lcao_elec.cpp]: The initialization has been added to the file

File: source/module_esolver/orb_paramter_init.cpp (hunk 8 above)
Comment (A-006) [reply source/module_esolver/orb_paramter_init.cpp]: The const has been added to the file

File: source/module_esolver/esolver_ks_lcao.h (hunk 6 above)
Comment (A-006) [reply source/module_esolver/esolver_ks_lcao.h]: The file location has been changed

File: source/module_hamilt_lcao/module_gint/grid_technique.cpp [hunk 9]
```diff
@@ -111,24 +32,12 @@ Grid_Technique::~Grid_Technique()
...
-                                  const int& startz_current,
-                                  const UnitCell& ucell,
-                                  const LCAO_Orbitals& orb,
-                                  const int num_stream)
+void Grid_Technique::set_pbc_grid(const int& ncx_in, const int& ncy_in, const int& ncz_in, const int& bx_in, const int& by_in,
+                      const int& bz_in, const int& nbx_in, const int& nby_in, const int& nbz_in, const int& nbxx_in,
+                      const int& nbzp_start_in, const int& nbzp_in, const int& ny, const int& nplane,
+                      const int& startz_current, const UnitCell& ucell, const double& dr_uniform, 
+                      std::vector<double>& rcuts,std::vector<std::vector<double>>& psi_u,std::vector<std::vector<double>>& dpsi_u,
+                      std::vector<std::vector<double>>& d2psi_u,const int& num_stream)
```
Comment (dzzz2001) [inline source/module_hamilt_lcao/module_gint/grid_technique.cpp]: add const

**Review by mohanchen — APPROVED** [review]

**Pull request merged**